src/qt/transactiontablemodel.cpp

```diff
@@ -131,7 +136,12 @@ class TransactionTablePriv
             }
             if(showTransaction)
             {
-                LOCK2(cs_main, wallet->cs_wallet);
+                TRY_LOCK(cs_main,lock_main);
+                if (!lock_main)
+                    return;
```

Comment:
This (and many other lines like this) may break transaction list completely. I'd suggest some other approach (caching list and atomically adding all the missing txs?)